### PR TITLE
NO-JIRA: task graph: comments and cleanups

### PR DESCRIPTION
**Pull request by petr-muller**
I want to eventually resolve [OCPBUGS-22442](https://issues.redhat.com/browse/OCPBUGS-22442) but this code is ancient and it took me a while to understand it. While I was grokking I made several opportunistic improvements which hopefully simplify the code and make it easier to understand.

### CODE REVIEW

File: pkg/payload/task_graph.go [hunk 1]
```diff
@@ -487,9 +492,10 @@ func RunGraph(ctx context.Context, graph *TaskGraph, maxParallelism int, fn func
 	}
 
 	var inflight int
-	nextNode := getNextNode()
-	done := false
+	var done bool
+
 	for !done {
+		nextNode := getNextNode()
```
Comment (petr-muller) [inline pkg/payload/task_graph.go:498]: Doing this at the start of the cycle is equivalent to doing it once before the cycle + at the end of each cycle if !done

File: pkg/payload/task_graph_test.go [hunk 2]
```diff
@@ -703,21 +703,11 @@ func TestRunGraph(t *testing.T) {
...
-{
-  "apiVersion": "v1",
-  "kind": "ConfigMap",
-  "metadata": {
-    "name": "%s",
-    "namespace": "default"
-  }
-}
-`, name)))
-			if err != nil {
```
Comment (petr-muller) [inline pkg/payload/task_graph_test.go:717]: hide the irrelevant visual clutter

**Review by wking — APPROVED** [review]
/lgtm

**Comment by wking** [comment]
Pure refactor, no intention of backports, so:

/label no-qe

**Comment by wking** [comment]
[Unstable node][1] is unrelated to this refactor.

/override ci/prow/e2e-agnostic-ovn-upgrade-out-of-change

[1]: https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_cluster-version-operator/1099/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-ovn-upgrade-out-of-change/1851698069918191616

**Pull request merged**